Backport handling of multiple initrd paths, like those needed by Manjaro
installations. Note, it still requires resolution of
https://savannah.gnu.org/bugs/index.php?47681 on grub's end.

Commits: https://salsa.debian.org/installer-team/os-prober/-/commit/7641c2da0c81f78c5f2ee2a66a1c21350cab03fc
         https://salsa.debian.org/installer-team/os-prober/-/commit/53b920e106f13acf87ef8a275161e20f94feeb8a

--- a/linux-boot-probes/mounted/common/40grub2
+++ b/linux-boot-probes/mounted/common/40grub2
@@ -78,11 +78,21 @@ parse_grub_menu () {
				fi
			;;
			initrd)
-				initrd="$(echo "$2" | sed 's/(.*)//')"
-				# Initrd same.
-				if [ "$partition" != "$bootpart" ]; then
-					initrd="/boot$initrd"
-				fi
+				shift
+				initrd=""
+				for initrd_path in "$@"; do
+					# sed hack, as above
+					initrd_path="$(echo "$initrd_path" | sed 's/(.*)//')"
+					# Initrd same.
+					if [ "$partition" != "$bootpart" ]; then
+						initrd_path="/boot$initrd_path"
+					fi
+					if [ -z "$initrd" ]; then
+						initrd="$initrd_path"
+					else
+						initrd="$initrd $initrd_path"
+					fi
+				done
			;;
			"}")
				entry_result
